pkg/resource/finalizer.go

```diff
@@ -46,3 +46,26 @@ func (fins *Finalizers) Remove(fin Finalizer) bool {
 func (fins Finalizers) Empty() bool {
 	return len(fins) == 0
 }
+
+// Has returns true if fin is present in the list of finalizers.
+func (fins Finalizers) Has(fin Finalizer) bool {
+	for _, f := range fins {
+		if f == fin {
+			return true
+		}
+	}
+
+	return false
+}
+
+// Set copies the finalizers from the other.
+func (fins *Finalizers) Set(other Finalizers) {
```

Comment:
`CopyFrom` maybe? `Set` looks weird when there is `Has` which accepts only one `Finalizer`.

Reply:
as `Finalizers` is a kind of a set (but actually a slice), I think it is consistent. I think `Copy` in Go is more of a top-level function, rather than a function on a type.

Reply:
Maybe it's a good idea to make `Copy(dst, src *Finalizers)`. But I don't think I have a strong opinion about this.